components/browser/state/src/main/java/mozilla/components/browser/state/action/BrowserAction.kt

```diff
@@ -16,27 +16,46 @@ import mozilla.components.lib.state.Action
  * [Action] implementation related to [BrowserState].
  */
 sealed class BrowserAction : Action
+
 /**
  * [BrowserAction] implementations related to updating the list of [TabSessionState] inside [BrowserState].
  */
 sealed class TabListAction : BrowserAction() {
     /**
      * Adds a new [TabSessionState] to the list.
+     *
+     * @property tab the [TabSessionState] to add
+     * @property select whether or not to the tab should be selected.
+     * @property parent the parent [TabSessionState] of the provided tab. The
+     * tab will be added next to its parent, if provided.
      */
-    data class AddTabAction(val tab: TabSessionState, val select: Boolean = false) : TabListAction()
+    data class AddTabAction(
+        val tab: TabSessionState,
+        val select: Boolean = false,
+        val parent: TabSessionState? = null
+    ) : TabListAction()
```

Comment:
I wonder if we even need to add the `parent` parameter here - since the `TabSessionState` can already contain a parent id. We could just use that (if it is set) and make sure that this is a valid parent - what do you think?

The higher-level use case for adding a tab would hide most of that anyways, I assume.

Reply:
Interesting. I was wondering about that too and thought it will make things easier for the callers. However, I just looked through the existing call sites and it makes not difference at all. I'll remove it from the action and just keep the parentId in the state.